### ISSUE TITLE
(tsan, c++, linux) tests.bins/tsan/qps_openloop_test flake

### ISSUE THREAD

**Issue by stanley-cheung** [grpc/grpc#6095]
https://grpc-testing.appspot.com/job/gRPC_master/16287/config=tsan,language=c++,platform=linux/console

probably a flake, green build before and after

```
PASSED: bins/tsan/channel_arguments_test:ChannelArgumentsTest.SetString
PASSED: bins/tsan/auth_property_iterator_test:AuthPropertyIteratorTest.DefaultCtor
FAILED: bins/tsan/qps_openloop_test
PASSED: bins/tsan/sync_unary_ping_pong_test
```

**Comment by jtattermusch** [comment]
TIMEOUT: bins/tsan/qps_openloop_test [pid=19149]    seen several times during the last day or so.

**Comment by vjpai** [comment]
Seen again on 4/12.
https://grpc-testing.appspot.com/job/gRPC_master/16444/config=tsan,language=c++,platform=linux/testReport/junit/(root)/tests/bins_tsan_qps_openloop_test/

**Comment by stanley-cheung** [comment]
https://grpc-testing.appspot.com/job/gRPC_master/16467/config=tsan,language=c++,platform=linux/

**Comment by sreecha** [comment]
Still happening as of: https://grpc-testing.appspot.com/job/gRPC_master/17753

**Comment by vjpai** [comment]
This race will be fixed by #7204 when it is merged.

**Comment by adelez** [comment]
I'm still seeing this failures on master: https://grpc-testing.appspot.com/job/gRPC_master/config=tsan,language=c++,platform=linux/18514/#showFailuresLink
